lib/endpoints/class-wp-json-users-controller.php

```diff
@@ -439,16 +437,36 @@ public function prepare_item_for_response( $user, $request ) {
 		$context = ! empty( $request['context'] ) ? $request['context'] : 'embed';
 		$data = $this->filter_response_by_context( $data, $context );
 
-		$data['_links'] = array(
-			'self'     => array(
-				'href' => json_url( '/wp/users/' . $user->ID ),
+		// Wrap the data in a response object
+		$data = json_ensure_response( $data );
+
+		$links = $this->prepare_links( $user );
+		foreach ( $links as $rel => $attributes ) {
+			$other = $attributes;
+			unset( $other['href'] );
+			$data->add_link( $rel, $attributes['href'], $other );
+		}
+
+		return apply_filters( 'json_prepare_user', $data, $user, $request );
+	}
+
+	/**
+	 * Prepare links for the request.
+	 *
+	 * @param WP_Post $user User object.
+	 * @return array Links for the given user.
+	 */
+	protected function prepare_links( $user ) {
+		$links = array(
+			'self' => array(
+				'href' => json_url( sprintf( '/wp/users/%d', $user->ID ) ),
 			),
-			'archives' => array(
-				'href' => json_url( '/wp/users/' . $user->ID . '/posts' ),
```

Comment:
@danielbachhuber the archives link never worked in v1.0, and has an enhancement issue #843